src/socket-client.ts

```diff
@@ -46,10 +47,21 @@ export class SocketClient extends EventEmitter {
     }
 
     private static completeSocketOptions(options: Partial<Options> = {}): Options {
-        return {
+        const mergedOptions = {
             ...SocketClient.createDefaultSocketOptions(),
             ...options
+        };
+
+        /**
+         * If no explicit polling or websockets flag was set,
+         * decide implicitly on whether to force websockets
+         * based on the runtime enrivonment.
+         */
+        if (!options.forceWebsockets && !options.disableWebsockets) {
```

Comment:
this should be OR

Reply:
No this looks correct, only use the helper function if both options aren't set.